xray/client/xray_viewer.ts

```diff
@@ -98,23 +107,22 @@ export class XRayViewer {
       edge_length_px_for_level *= 2;
       level += 1;
     }
+    this.displayLevel = level;
 
     // We encode the matrix column major.
```

Comment:
This comment probably belongs closer to the `matrixToString` call. Consider doing this:

```
const matrixStr = matrixToString(matrix);  // Note: matrix is encoded column-major.
window.fetch(
    ...
)
```

Reply:
done.